src/utils/googleTagManager.ts

```diff
@@ -171,3 +172,34 @@ export const trackEvent = ({
     dataLayer,
   })
 }
+
+export const trackSafeAppEvent = ({
+  event,
+  name,
+  method,
+  params,
+  sdkVersion,
+}: {
+  event: GTM_EVENT
+  name: string
+  method: string
+  params: any
+  sdkVersion: string
+}): void => {
+  const dataLayer = {
+    event,
+    chainId: _getChainId(),
+    safeAppName: name,
+    safeAppMethod: method,
+    safeAppParams: params,
+    safeAppSDKVersion: sdkVersion,
+  }
+
+  if (!IS_PRODUCTION) {
+    console.info('[GTM]', dataLayer)
+  }
+
+  TagManager.dataLayer({
+    dataLayer,
+  })
```

Comment:
Duplicate code with trackEvent. Should be abstracted.

Reply:
yes, will take a look. As I said is a draft. I'm testing here and there how it works